sdk/core/azure-core/src/main/java/com/azure/core/implementation/ImplUtils.java

```diff
@@ -347,6 +350,16 @@ public static URL createUrl(String urlString) throws MalformedURLException {
         return new URL(urlString);
     }
 
+    public static <T> Class<? extends T> getClassByName(String className, Class<T> base) {
+        Objects.requireNonNull("'className' cannot be null");
+        try {
+            return Class.forName(className, false, ImplUtils.class.getClassLoader()).asSubclass(base);
```

Comment:
Why do we call `asSubclass`? Mind adding some documentation on the reasoning

Reply:
Reading the PR further it seems we are actually casting to a super class, instead should this be doing an unchecked cast of

```java
return (Class<? extends T>) Class.forName(className, false, ImplUtils.class.getClassLoader());
```